cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AuthenticationUtils.java

```diff
@@ -41,32 +44,33 @@
     /**
      * Validates Kafka client authentication for all components based on Apache Kafka clients.
      *
+     * @param reconciliation The reconciliation
      * @param authentication    The authentication object from CRD
      * @param tls   Indicates whether TLS is enabled or not
      */
     @SuppressWarnings("BooleanExpressionComplexity")
-    public static void validateClientAuthentication(KafkaClientAuthentication authentication, boolean tls)    {
+    public static void validateClientAuthentication(Reconciliation reconciliation, KafkaClientAuthentication authentication, boolean tls)    {
```

Comment:
For this, because it's a validation method that throws, we could perhaps return a warning message to be logged by the caller (when non-null)? That doesn't work for the places where we currently log and then throw, but do we really need to log in those places? Shouldn't the catcher of the exception be the one to log?

Reply:
I agree that the catch block should log when an exception is thrown. What about cases, where no exception is thrown but some debug/warn message is needed to be logged?

Reply:
That's where returning the warning message would let the caller do the logging. It was just a thought.